pkg/frontend/querymiddleware/astmapper/sharding.go

```diff
@@ -26,11 +26,13 @@ func NewSharding(shardSummer ASTMapper) ASTMapper {
 	)
 }
 
-type Squasher = func(...parser.Expr) (parser.Expr, error)
+type Squasher = func(...EmbeddedQuery) (parser.Expr, error)
 
 type ShardLabeller interface {
 	GetLabelName() string
 	GetLabelValue(shard int) string
```

Comment:
Do we still need these as part of the `ShardLabeller` interface if we have `GetLabelMatcher`?

Reply:
True that they're not actually needed anymore with `GetLabelMatcher`, I just wanted to keep them because I thought it was clearer, but I can remove them if you think that'd be better.

Reply:
If we only use `GetLabelMatcher` now, then I would remove the other methods.